src/gam.py

```diff
@@ -7847,73 +7838,115 @@ def _formatOAuth2ServiceData(private_key, private_key_id):
   key_json['private_key_id'] = private_key_id
   return json.dumps(key_json, indent=2, sort_keys=True)
 
-def showServiceAccountKeys():
+def doShowServiceAccountKeys():
   iam = buildGAPIServiceObject('iam', None)
+  keyTypes = 'USER_MANAGED'
+  i = 3
+  while i < len(sys.argv):
+    myarg = sys.argv[i].lower().replace('_', '')
+    if myarg == 'all':
+      keyTypes = None
+      i += 1
+    elif myarg in ['system', 'systemmanaged']:
+      keyTypes = 'SYSTEM_MANAGED'
+      i += 1
+    elif myarg in ['user', 'usermanaged']:
+      keyTypes = 'USER_MANAGED'
+      i += 1
+    else:
+      controlflow.system_error_exit(3, '%s is not a valid argument to "gam show sakeys"' % myarg)
   name = 'projects/-/serviceAccounts/%s' % GM_Globals[GM_OAUTH2SERVICE_ACCOUNT_CLIENT_ID]
-  result = gapi.call(iam.projects().serviceAccounts().keys(), 'list', name=name)
-  keys = result.get('keys')
-  if not keys or len(keys) == 0:
+  currentPrivateKeyId = GM_Globals[GM_OAUTH2SERVICE_JSON_DATA]['private_key_id']
+  keys = gapi.get_items(iam.projects().serviceAccounts().keys(), 'list', 'keys',
+                        name=name, keyTypes=keyTypes)
+  if not keys:
     print('No keys')
     return
   parts = keys[0]['name'].rsplit('/')
-  i = 0
-  while i < len(parts):
-    if parts[i] not in ['keys']:
-      print('{0}: {1}'.format(parts[i], parts[i+1]))
-    i += 2
+  for i in range(0, 4, 2):
+    print('{0}: {1}'.format(parts[i][:-1], parts[i+1]))
   for key in keys:
-    key['name'] = key.get('name').rsplit('/', 1)[-1]
-  print_json(None, keys)
+    keyName = key['name'].rsplit('/', 1)[-1]
+    print('{0}: {1}, Current: {2}'.format('name', keyName, keyName == currentPrivateKeyId))
+    for field in ['keyAlgorithm', 'keyOrigin', 'keyType', 'validAfterTime', 'validBeforeTime']:
```

Comment:
the problem with this approach is it will exclude any new fields/attributes the API adds at a later time.